src/clj/clojure/core.clj

```diff
@@ -6861,46 +6861,88 @@
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;; clojure version number ;;;;;;;;;;;;;;;;;;;;;;
 
-(let [properties (with-open [version-stream (.getResourceAsStream
-                                             (clojure.lang.RT/baseLoader)
-                                             "clojure/version.properties")]
-                   (doto (new java.util.Properties)
-                     (.load version-stream)))
-      version-string (.getProperty properties "version")
-      [_ major minor incremental qualifier snapshot]
-      (re-matches
-       #"(\d+)\.(\d+)\.(\d+)(?:-([a-zA-Z0-9_]+))?(?:-(SNAPSHOT))?"
-       version-string)
-      clojure-version {:major       (Integer/valueOf ^String major)
-                       :minor       (Integer/valueOf ^String minor)
-                       :incremental (Integer/valueOf ^String incremental)
-                       :qualifier   (if (= qualifier "SNAPSHOT") nil qualifier)}]
-  (def ^:dynamic *clojure-version*
-    (if (.contains version-string "SNAPSHOT")
-      (clojure.lang.RT/assoc clojure-version :interim true)
-      clojure-version)))
-
-(add-doc-and-meta *clojure-version*
-  "The version info for Clojure core, as a map containing :major :minor 
-  :incremental and :qualifier keys. Feature releases may increment 
-  :minor and/or :major, bugfix releases will increment :incremental. 
-  Possible values of :qualifier include \"GA\", \"SNAPSHOT\", \"RC-x\" \"BETA-x\""
-  {:added "1.0"})
+(def
+  ^{:dynamic    true
+    :deprecated "FIXME"
+    :added      "1.0"}
+  *clojure-version*
+  "DEPRECATED: replaced by *jaunt-version*
+
+  The version of Clojure from which this Jaunt build is derived. Version information is encoded as a
+  map containing :major :minor :incremental and :qualifier keys."
+  {:major       1
+   :minor       8
+   :incremental 0
+   :qualifier   nil})
+
+(defn clojure-version
+  "DEPRECATED: replaced by jaunt-version
+
+  Returns clojure version as a printable string."
+  {:added      "1.0"
+   :deprecated "FIXME"}
+  []
+  "1.8.0")
+
+(defn- parse-version [v]
+  {:pre [(string? v)]}
+  (let [[_ major minor incremental qualifier]
+        ,,(re-matches #"^(\d+)\.(\d+)\.(\d+)(-.*)?$" v)]
+    {:major        (Integer/valueOf ^String major)
+     :minor        (Integer/valueOf ^String minor)
+     :incremental  (Integer/valueOf ^String incremental)
+     :interim      (.endsWith ^String qualifier "-SNAPSHOT")
+     :qualifiers   (->> (.split ^String qualifier "-")
+                        (filter (complement empty?))
```

Comment:
No problem here, but I'm curious: any issues against using `not-empty` instead of composing with `complement`? I would guess it's because it doesn't look like a predicate.